code.py

```diff
@@ -8,64 +8,85 @@
 from audiomp3 import MP3Decoder
 import microcontroller
 
-audio = audiobusio.I2SOut(board.GP10, board.GP11, board.GP9) #BCLK_PIN, WS_PIN, SDIN_PIN
 server = HTTPServer(socketpool.SocketPool(wifi.radio))
+# IO setup
 board_led = digitalio.DigitalInOut(board.LED)
 board_led.direction = digitalio.Direction.OUTPUT
+lamp = pwmio.PWMOut(board.GP17)
 
-TARDIS_COMMANDS = ["demat", "mat", "flight", "cloister", "quickdemat", "quickmat", ]
+#Fetch list of sounds to be played
+sounds = os.listdir("sounds")
 
 def selectSound(command):
-    """Takes string command and checks if it is in `TARDIS_COMMANDS`, returning the path to the `.mp3` file if True
+    """Takes string command and checks if it is in `sounds`, returning the path to the `.mp3` file if True
     """
-    if command not in TARDIS_COMMANDS:
-        return False
-    return "sounds/%s.mp3" % command
+    sound = "{}.mp3".format(command)
+    if sound not in sounds:
+        return None
+    return "sounds/{}".format(sound)
 
-def playSound(sound):
+def playSound(sound, playLight=True):
+    global decoder
     """Play the given sound and run the lamp 
     :param sound: the path to the audio file to be played
     """
-    if audio.playing:
-        return None
-    else:
-        audio.play(MP3Decoder(sound))
-        asyncio.run(light())
-        return True
+    with audiobusio.I2SOut(board.GP10, board.GP11, board.GP9) as audio:#BCLK_PIN, WS_PIN, SDIN_PIN
+        if audio.playing:
+            return False
+        else:
+            decoder.file = open(sound, "rb")

```

Comment:
Saving the allocation overhead on `MP3Decoder`s is a good move, but we're not explicitly `close`ing the files. `decoder.file` should always be initialised here, so you should call `.close()` before opening a new file.

```suggestion
            decoder.file.close()
            decoder.file = open(sound, "rb")
```